### ISSUE TITLE
eigmax() and eigmin() values swapped with SymTridiagonal

### ISSUE THREAD

**Issue by jamiecjx** [JuliaLang/julia#48514]
I am using Julia Version 1.8.5

The output of eigmax() and eigmin() are swapped here for a 2x2 SymTridiagonal matrix.
This doesn't seem to occur if the SymTridiagonal is anything except 2x2. It also doesn't seem to occur for all 2x2 SymTridiagonals.

```julia
julia> using LinearAlgebra

julia> a = SymTridiagonal([-1.26189, 1.17464],[0.98587])
2×2 SymTridiagonal{Float64, Vector{Float64}}:
 -1.26189  0.98587
  0.98587  1.17464

julia> eigmin(a)
1.5235729029864098

julia> eigmin(Array(a))
-1.6108229029864096

julia> eigmax(a)
-1.6108229029864096

julia> eigmax(Array(a))
1.5235729029864098
```

**Comment by jishnub** [comment]
I wonder if the LAPACK call guarantees the ordering that is assumed in these functions?
```julia
julia> LAPACK.stegr!('N', 'I', copy(a).dv, copy(a).ev, 0.0, 0.0, 2, 2)
([-1.6108229029864096], [6.91663303144973e-310;;])

julia> LAPACK.stegr!('N', 'I', copy(a).dv, copy(a).ev, 0.0, 0.0, 1, 1)
([1.5235729029864098], [6.9165579122066e-310;;])

julia> LAPACK.stegr!('N', 'I', copy(a).dv, copy(a).ev, 0.0, 0.0, 1, 2)
([-1.6108229029864096, 1.5235729029864098], [6.9166329892672e-310 6.9166329804515e-310])
```

**Comment by jishnub** [comment]
Interestingly, python seems to get this right:
```python
In [1]: import numpy as np

In [4]: from scipy import linalg

In [5]: scipy.linalg.eigh_tridiagonal(np.array([-1.26189, 1.17464]), np.array([0.98587]), eigvals_only=True, select='i', select_range=[0,1])
Out[5]: array([-1.6108229,  1.5235729])

In [6]: scipy.linalg.eigh_tridiagonal(np.array([-1.26189, 1.17464]), np.array([0.98587]), eigvals_only=True, select='i', select_range=[1,1])
Out[6]: array([1.5235729])

In [7]: scipy.linalg.eigh_tridiagonal(np.array([-1.26189, 1.17464]), np.array([0.98587]), eigvals_only=True, select='i', select_range=[0,0])
Out[7]: array([-1.6108229])
```

**Comment by aravindh-krishnamoorthy** [comment]
Hello, this issue was lying around for a while and not fixed in LAPACK 3.11.0, hence, I took the liberty to file an issue with LAPACK here: https://github.com/Reference-LAPACK/lapack/issues/826

Due credits are given. I'll try to follow this up and get it fixed (or fix it).

**Comment by aravindh-krishnamoorthy** [comment]
UPDATE: A minimum working example (MWE) is submitted for the related LAPACK issue: https://github.com/Reference-LAPACK/lapack/issues/826

**Comment by jishnub** [comment]
Thank you very much for this!

**Comment by aravindh-krishnamoorthy** [comment]
**Update:** This fix is included in LAPACK 3.12.0 released on 24.11.2023.
